servers/services/src/main/java/org/projectnessie/services/rest/TreeResourceWithAuthorizationChecks.java

```diff
@@ -100,63 +96,24 @@ public Reference getReferenceByName(String refName) throws NessieNotFoundExcepti
   @Override
   public Reference createReference(@Nullable String sourceRefName, Reference reference)
       throws NessieNotFoundException, NessieConflictException {
-    if (sourceRefName == null) {
-      sourceRefName = getConfig().getDefaultBranch();
-    }
-
     if (reference instanceof Branch) {
       getAccessChecker()
           .canCreateReference(createAccessContext(), BranchName.of(reference.getName()));
     } else if (reference instanceof Tag) {
       getAccessChecker().canCreateReference(createAccessContext(), TagName.of(reference.getName()));
-    } else {
```

Comment:
does this `else` block need to be removed?

Reply:
yes, the same check is happening in the super class. Also the purpose of `TreeResourceWithAuthorizationChecks` is to only perform authz checks and not validation logic (which was represented by this else part)

Reply:
yes, the same check is happening in the super class. Also the purpose of `TreeResourceWithAuthorizationChecks` is to only perform authz checks and not validation logic (which was represented by this else part)

Reply:
got it, thanks